src/index.test.ts

```diff
@@ -351,6 +354,44 @@ describe('post()', () => {
     // assert error is empty
     expect(error).toBe(undefined);
   });
+
+  it('request body type when optional', async() => {
+    const client = createClient<paths>();
+    const { body: mockData } = mockResponse(201, {status: 'success'});
+    const { data, error, response } = await client.post('/post/optional', {
+      body: {
+        title: '',
+        publish_date: 3,
+        body: ''
+      }
+    })
+
+    // assert correct data was returned
+    expect(data).toEqual(mockData);
```

Comment:
If this is a type-only test, then we don’t need to make additional assertions in this test that exist elsewhere; these can probably be deleted

Reply:
Also, if you could add comments on these tests explaining that the actual test is testing the typing on the `await client.post(…)`, that would be helpful. See other tests for reference.